src/urlValidator.js

```diff
@@ -37,8 +45,8 @@ module.exports = class {
       let ipv4Error = false;
       let ipv6Error = false;
       ipAddresses = [];
-      const ipv4Resolver = new Promise((resolve, reject) => dns.resolve4(hostname, (error, addresses) => error ? reject(error) : resolve(addresses)));
-      const ipv6Resolver = new Promise((resolve, reject) => dns.resolve6(hostname, (error, addresses) => error ? reject(error) : resolve(addresses)));
+      const ipv4Resolver = new Promise((resolve, reject) => handleRejection(() => dns.resolve4(hostname, (error, addresses) => error ? reject(error) : resolve(addresses)), reject));
```

Comment:
I'd suggest moving the whole line into the function, to reduce the complexity.  I.e. having the handleRejection not take in `reject` but instead have the code to both convert to a `new Promise` and `handle both sync and async` errors.

Reply:
Makes sense. I did that :)

Still it is not awesome. it triggers https://eslint.org/docs/rules/no-async-promise-executor which is what is actually happening. The `dns.resolve6` looks to be a "hybrid" function.

It can both return a rejected Promise, or return an error through the callback. So the function itself is a async promise executor in my understanding.

Is there something I am missing?

Reply:
The eslint rule is correct:

I don't think the current methods actually return promises, at least not according to the docs: https://nodejs.org/api/dns.html#dns_dns_resolve4_hostname_options_callback

So we don't need the `await` inside the try/catch (if it is sync, then the catch will reject, and if it is async, then the reject passed to the callback will get it).

However, instead there is a promised version of the library already:
* https://nodejs.org/api/dns.html#dns_dnspromises_resolve4_hostname_options

So we could use that.